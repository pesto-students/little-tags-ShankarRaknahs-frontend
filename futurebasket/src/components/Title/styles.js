import { makeStyles } from '@material-ui/core/styles';
import COLORS from '../../config/colors.config';

export const useStyles = makeStyles((theme) => ({
  title: {
    color: COLORS.TITLE1,
    fontWeight: 'bold',
    fontFamily: 'Poppins',
    fontSize: 25,
    [theme.breakpoints.up('sm')]: {
      display: 'block',
    },
    marginBottom: -20,
    marginInlineStart: -5,
  },
  title2: {
    color: COLORS.SECONDARY,
    fontWeight: 'bold',
    fontFamily: 'Poppins',
    fontSize: 25,
    [theme.breakpoints.up('sm')]: {
      display: 'block',
    },
  },
}));
