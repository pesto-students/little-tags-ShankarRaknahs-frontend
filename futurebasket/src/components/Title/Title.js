import { Typography } from '@material-ui/core';

import { useStyles } from './styles';

const Title = (title) => {
  const classes = useStyles();
  const firstName = title.firstName;
  const lastName = title.lastName;
  return (
    <div>
      <Typography className={classes.title} variant='h6' noWrap>
        {firstName}
      </Typography>

      <Typography className={classes.title2} variant='h6' noWrap>
        {lastName}
      </Typography>
    </div>
  );
};

export default Title;
