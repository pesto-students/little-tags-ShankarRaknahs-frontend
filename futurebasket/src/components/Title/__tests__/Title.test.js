import React from 'react';

import renderer from 'react-test-renderer';

// import { render } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';
import Title from '../Title';

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJSON from 'enzyme-to-json';
Enzyme.configure({ adapter: new Adapter() });
const { mount } = Enzyme;

describe('<Header />', () => {
  it('render correctly (snapshot)', () => {
    const app = { firstName: 'Future', lastName: 'basket' };
    const tree = renderer.create(<Title {...app} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });

  it('render correctly using Enzyme (snapshot)', () => {
    const app = { firstName: 'Future', lastName: 'basket' };
    const tree = mount(<Title {...app} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });
});
