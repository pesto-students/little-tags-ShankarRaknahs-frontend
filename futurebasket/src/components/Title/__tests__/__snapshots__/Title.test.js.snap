// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> render correctly (snapshot) 1`] = `undefined`;

exports[`<Title /> render correctly using Enzyme (snapshot) 1`] = `
<Title
  firstName="Future"
  lastName="basket"
>
  <div>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-1"
      noWrap={true}
      variant="h6"
    >
      <ForwardRef(Typography)
        className="makeStyles-title-1"
        classes={
          Object {
            "alignCenter": "MuiTypography-alignCenter",
            "alignJustify": "MuiTypography-alignJustify",
            "alignLeft": "MuiTypography-alignLeft",
            "alignRight": "MuiTypography-alignRight",
            "body1": "MuiTypography-body1",
            "body2": "MuiTypography-body2",
            "button": "MuiTypography-button",
            "caption": "MuiTypography-caption",
            "colorError": "MuiTypography-colorError",
            "colorInherit": "MuiTypography-colorInherit",
            "colorPrimary": "MuiTypography-colorPrimary",
            "colorSecondary": "MuiTypography-colorSecondary",
            "colorTextPrimary": "MuiTypography-colorTextPrimary",
            "colorTextSecondary": "MuiTypography-colorTextSecondary",
            "displayBlock": "MuiTypography-displayBlock",
            "displayInline": "MuiTypography-displayInline",
            "gutterBottom": "MuiTypography-gutterBottom",
            "h1": "MuiTypography-h1",
            "h2": "MuiTypography-h2",
            "h3": "MuiTypography-h3",
            "h4": "MuiTypography-h4",
            "h5": "MuiTypography-h5",
            "h6": "MuiTypography-h6",
            "noWrap": "MuiTypography-noWrap",
            "overline": "MuiTypography-overline",
            "paragraph": "MuiTypography-paragraph",
            "root": "MuiTypography-root",
            "srOnly": "MuiTypography-srOnly",
            "subtitle1": "MuiTypography-subtitle1",
            "subtitle2": "MuiTypography-subtitle2",
          }
        }
        noWrap={true}
        variant="h6"
      >
        <h6
          className="MuiTypography-root makeStyles-title-1 MuiTypography-h6 MuiTypography-noWrap"
        >
          Future
        </h6>
      </ForwardRef(Typography)>
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title2-2"
      noWrap={true}
      variant="h6"
    >
      <ForwardRef(Typography)
        className="makeStyles-title2-2"
        classes={
          Object {
            "alignCenter": "MuiTypography-alignCenter",
            "alignJustify": "MuiTypography-alignJustify",
            "alignLeft": "MuiTypography-alignLeft",
            "alignRight": "MuiTypography-alignRight",
            "body1": "MuiTypography-body1",
            "body2": "MuiTypography-body2",
            "button": "MuiTypography-button",
            "caption": "MuiTypography-caption",
            "colorError": "MuiTypography-colorError",
            "colorInherit": "MuiTypography-colorInherit",
            "colorPrimary": "MuiTypography-colorPrimary",
            "colorSecondary": "MuiTypography-colorSecondary",
            "colorTextPrimary": "MuiTypography-colorTextPrimary",
            "colorTextSecondary": "MuiTypography-colorTextSecondary",
            "displayBlock": "MuiTypography-displayBlock",
            "displayInline": "MuiTypography-displayInline",
            "gutterBottom": "MuiTypography-gutterBottom",
            "h1": "MuiTypography-h1",
            "h2": "MuiTypography-h2",
            "h3": "MuiTypography-h3",
            "h4": "MuiTypography-h4",
            "h5": "MuiTypography-h5",
            "h6": "MuiTypography-h6",
            "noWrap": "MuiTypography-noWrap",
            "overline": "MuiTypography-overline",
            "paragraph": "MuiTypography-paragraph",
            "root": "MuiTypography-root",
            "srOnly": "MuiTypography-srOnly",
            "subtitle1": "MuiTypography-subtitle1",
            "subtitle2": "MuiTypography-subtitle2",
          }
        }
        noWrap={true}
        variant="h6"
      >
        <h6
          className="MuiTypography-root makeStyles-title2-2 MuiTypography-h6 MuiTypography-noWrap"
        >
          basket
        </h6>
      </ForwardRef(Typography)>
    </WithStyles(ForwardRef(Typography))>
  </div>
</Title>
`;
